osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CheckServiceFunctionChainMetaTask.java

```diff
@@ -0,0 +1,157 @@
+		task.txBroadcastUtil = this.txBroadcastUtil;
+
+		return task;
+	}
+
+	@Override
+	public void executeTransaction(EntityManager em) throws Exception {
+		this.tg = new TaskGraph();
+		this.securityGroup = em.find(SecurityGroup.class, this.securityGroup.getId());
+
+		if (this.securityGroup.getNetworkElementId() == null && this.securityGroup.getServiceFunctionChain() != null) {
+			// if SFC binded and network element null -> create PC
+			this.portPairGroups = getPortPairGroupWithSameProjectId(em, this.securityGroup.getProjectId(),
+					this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+			this.tg.appendTask(this.createServiceFunctionChainTask.create(this.securityGroup, this.portPairGroups));
+
+		} else if (this.securityGroup.getNetworkElementId() != null && this.securityGroup.getServiceFunctionChain() != null) {
+			// if SFC binded and network element not null -> update PC
+			this.portPairGroups = getPortPairGroupWithSameProjectId(em, this.securityGroup.getProjectId(),
+					this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+			try (SdnRedirectionApi sdnApi = this.apiFactoryService
+					.createNetworkRedirectionApi(this.securityGroup.getVirtualizationConnector())) {
+
+				List<NetworkElement> sfcPortPairGroups = sdnApi
+						.getNetworkElements(new NetworkElementImpl(this.securityGroup.getNetworkElementId()));
+
+				if (!this.portPairGroups.equals(sfcPortPairGroups)) {
+					this.tg.appendTask(
+							this.updateServiceFunctionChainTask.create(this.securityGroup, this.portPairGroups));
+				}
+			}
+
+		} else if (this.securityGroup.getNetworkElementId() != null && this.securityGroup.getServiceFunctionChain() == null) {
+			// if SFC not binded and network element not null -> delete PC
+			this.tg.appendTask(this.deleteServiceFunctionChainTask.create(this.securityGroup));
+
+		}
+	}
+
+	static List<NetworkElement> getPortPairGroupWithSameProjectId(EntityManager em, String projectId,
+			List<VirtualSystem> vsList) throws VmidcBrokerValidationException {
+
+		List<NetworkElement> portPairGroups = new ArrayList<>();
+
+		for (VirtualSystem vs : vsList) {
+			if (vs.getDeploymentSpecs().isEmpty()) {
+				throw new VmidcBrokerValidationException(
+						String.format("No deployment found for virtual system '%s'.", vs.getName()));
+			}
+
+			// Currently, We have no information on OSC SG about the OST region. SFC do not support port chain with multiple region deployment.
+			// We cannot create Port Chain of VNF Deployment's belonging to different region. If we find the VS with multiple DS on different region throw an exception.
+			List<DeploymentSpec> dsList = DeploymentSpecEntityMgr
+					.findDeploymentSpecsByVirtualSystemProjectWithDefaultRegionOne(em, vs, projectId);
+			if (dsList.size() > 1) {
+				throw new VmidcBrokerValidationException(String
+						.format("Multiple deployments found for vs '%s' belonging to the same project.", vs.getName()));
+			}
+
+			if (!dsList.isEmpty()) {
```

Comment:
if dslist is empty we need to throw an exception right specifying there are no deployments for the binded chain.

Reply:
We already have it.

Reply:
can you point me to the code?

Reply:
can you point me to the code?

I see if the ds's PortGroupId is set to null/empty we throw the exception...but not for the case when there are 0 ds'es

Reply:
https://github.com/larkinscarvalho/osc-core/blob/sfc/osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CheckServiceFunctionChainMetaTask.java#L115

Reply:
We check for each vs if we have deployment specs

Reply:
Having a DS is not enough though. you have to have a "matching" ds. I would remove that check and use the list you get from findDeploymentSpecsByVirtualSystemProjectWithDefaultRegionOne